src/core/qgsofflineediting.cpp

```diff
@@ -834,7 +839,16 @@ QgsVectorLayer *QgsOfflineEditing::copyVectorLayer( QgsVectorLayer *layer, sqlit
       QgsAttributes newAttrs( containerType == GPKG ? attrs.count() + 1 : attrs.count() );
       for ( int it = 0; it < attrs.count(); ++it )
       {
-        newAttrs[column++] = attrs.at( it );
+        QVariant attr = attrs.at( it );
+        if ( layer->fields().at( it ).type() == QVariant::StringList )
+        {
+          attr = attr.toStringList().join( ',' );
```

Comment:
Won't you need to escape "," values in the original strings here?

Reply:
@nyalldawson , right, I was going to add this as a comment. Escaping "," values isn't the most user friendly thing there is, but I guess this could be explained in documentation to users.

As far as _reading_ goes, it doesn't matter that much (and if you don't edit the field in the offline layer, it won't cause an issue when synchronizing back).

Reply:
I was meaning you should automatically escape them here, and unescape at decode time..